lib/vmpooler/pool_manager.rb

```diff
@@ -292,6 +292,100 @@ def destroy_vm(vm, pool)
       end
     end
 
+    def create_vm_snapshot(vm, snapshot_name)
+      Thread.new do
+        _create_vm_snapshot(vm, snapshot_name)
+      end
+    end
+
+    def _create_vm_snapshot(vm, snapshot_name)
```

Comment:
What sort of scrutiny do you have in mind?  As-is the only method calling `create_vm_snapshot` is the API, which is generating the `snapshot_name` var right before calling this method.